src/compute/src/lib.rs

```diff
@@ -28,98 +28,121 @@ pub mod memory_management;
-    pub host: String,
+    #[clap(long = "host")]
+    pub listen_addr: Option<String>,
 
-    /// The address of the compute node's meta client.
-    ///
-    /// Optional, we will use listen_address if not specified.
     #[clap(long)]
     pub client_address: Option<String>,
 
-    #[clap(long, default_value = "hummock+memory")]
-    pub state_store: String,
+    #[clap(long)]
+    pub state_store: Option<String>,
 
-    #[clap(long, default_value = "127.0.0.1:1222")]
-    pub prometheus_listener_addr: String,
+    #[clap(long)]
+    pub prometheus_listener_addr: Option<String>,
 
-    /// Used for control the metrics level, similar to log level.
-    /// 0 = close metrics
-    /// >0 = open metrics
-    #[clap(long, default_value = "0")]
-    pub metrics_level: u32,
+    #[clap(long)]
+    pub metrics_level: Option<u32>,
 
-    #[clap(long, default_value = "http://127.0.0.1:5690")]
-    pub meta_address: String,
+    #[clap(long)]
+    pub meta_address: Option<String>,
 
-    /// Enable reporting tracing information to jaeger.
     #[clap(long)]
-    pub enable_jaeger_tracing: bool,
+    pub enable_jaeger_tracing: Option<bool>,
 
-    /// Enable async stack tracing for risectl.
-    #[clap(long, arg_enum, default_value_t = AsyncStackTraceOption::On)]
-    pub async_stack_trace: AsyncStackTraceOption,
+    #[clap(long, arg_enum)]
+    pub async_stack_trace: Option<AsyncStackTraceOption>,
 
-    /// Path to file cache data directory.
-    /// Left empty to disable file cache.
-    #[clap(long, default_value = "")]
-    pub file_cache_dir: String,
+    #[clap(long)]
+    pub file_cache_dir: Option<String>,
 
-    /// Endpoint of the connector node
+    /// Default value is read from the 'CONNECTOR_RPC_ENDPOINT' environment variable.
     #[clap(long, env = "CONNECTOR_RPC_ENDPOINT")]
     pub connector_rpc_endpoint: Option<String>,
 
+    #[clap(long)]
+    pub total_memory_bytes: Option<usize>,
+
+    #[clap(long)]
+    pub parallelism: Option<usize>,
+
     /// The path of `risingwave.toml` configuration file.
     ///
     /// If empty, default configuration values will be used.
-    ///
-    /// Note that internal system parameters should be defined in the configuration file at
-    /// [`risingwave_common::config`] instead of command line arguments.
     #[clap(long, default_value = "")]
     pub config_path: String,
+}
 
-    /// Total available memory in bytes, used by LRU Manager
-    #[clap(long, default_value_t = default_total_memory_bytes())]
-    pub total_memory_bytes: usize,
-
-    /// The parallelism that the compute node will register to the scheduler of the meta service.
-    #[clap(long, default_value_t = default_parallelism())]
-    pub parallelism: usize,
+impl OverwriteConfig for ComputeNodeOpts {
+    fn overwrite(self, config: &mut RwConfig) {
+        let c = &mut config.compute_node;
+        if let Some(v) = self.listen_addr {
+            c.listen_addr = v;
+        }
```

Comment:
This'll be really error-prone. 🥵 Suggest `serfig` or the proc-macro way.

Reply:
Yes, I'm replacing this with serfig 🥵